RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -1532,6 +1533,13 @@ class DeepTauId : public deep_tau::DeepTauBase {
                 << "invalid prediction = " << pred << " for tau_index = " << tau_index << ", pred_index = " << k;
           predictions.matrix<float>()(tau_index, k) = pred;
         }
+      } else {
+        for (int k = 0; k < deep_tau::NumberOfOutputs; ++k) {
+          if (k == 2)
```

Comment:
It's been hardcoded this way initially because k == 2 is the index for the tau in the [list of outputs](https://github.com/cms-tau-pog/cmssw/blob/ad3236873b5fe6395b89de6eb3981357c1767d96/RecoTauTag/RecoTau/plugins/DeepTauId.cc#L1106). With the rest set to 2, it ended up as -1. in [this calculation](https://github.com/cms-tau-pog/cmssw/blob/ad3236873b5fe6395b89de6eb3981357c1767d96/RecoTauTag/RecoTau/src/DeepTauBase.cc#L68-L69). If you have a preference for a different way of setting defaults, this can be changed

Reply:
The default can change in case `deep_tau::NumberOfOutputs` has different value. But maybe it is not a problem. Anyway, I would add to the code a comment explaining this for reference.